packages/typescript-estree/src/create-program/createProjectService.ts

```diff
@@ -123,38 +124,26 @@ export function createProjectService(
 
   if (options.defaultProject) {
     log('Enabling default project: %s', options.defaultProject);
-    let configRead;
+    let configFile: ts.ParsedCommandLine;
 
     try {
-      configRead = tsserver.readConfigFile(
+      configFile = getParsedConfigFile(
+        tsserver,
         options.defaultProject,
-        system.readFile,
+        path.dirname(options.defaultProject),
       );
     } catch (error) {
       throw new Error(
-        `Could not parse default project '${options.defaultProject}': ${(error as Error).message}`,
-      );
-    }
-
-    if (configRead.error) {
-      throw new Error(
-        `Could not read default project '${options.defaultProject}': ${tsserver.formatDiagnostic(
-          configRead.error,
-          {
-            getCurrentDirectory: system.getCurrentDirectory,
-            getCanonicalFileName: fileName => fileName,
-            getNewLine: () => os.EOL,
-          },
-        )}`,
+        `Could not read default project '${options.defaultProject}': ${(error as Error).message}`,
       );
     }
 
     service.setCompilerOptionsForInferredProjects(
-      (
-        configRead.config as {
-          compilerOptions: ts.server.protocol.InferredProjectCompilerOptions;
-        }
-      ).compilerOptions,
+      // NOTE: The inferred projects API is not intended for source files when a tsconfig
+      // exists.  There is no API that generates an InferredProjectCompilerOptions suggesting
+      // it is meant for hard coded options passed in.  Hard casting as a work around.
```

Comment:
Super teeny nitpicky nit: I see "casting" as a runtime thing. "Asserting" is what we do in the type system. 

```suggestion
      // it is meant for hard coded options passed in. Hard asserting as a work around.
```